### PR TITLE
Maintenance Update

### PR DESCRIPTION
**Pull request by bkendall**
Various updates.

- add documentation generator w/ `jsdoc`
- add documentation generation script w/ upload to GitHub Pages
- clean up formatting in code/tests
- clean up docs
- add link to documentation in README

### CODE REVIEW

**Comment by podviaznikov** [comment]
I like it: http://runnable.github.io/ponos/!
+1

**Comment by bkendall** [comment]
For the future: we can [document throws](http://usejsdoc.org/tags-throws.html)

File: resources/generate-docs.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
```
Comment (anandkumarpatel) [inline resources/generate-docs.sh]: when does this script get run?

Comment (bkendall) [inline resources/generate-docs.sh]: @anandkumarpatel manually for now. could be added to a `postpublish` hook w/ `npm`, but I've not looked into those before, so just going one step at a time.

**Comment by rsandor** [comment]
:+1:

**Pull request merged**